File: June Challenge 2020/07 coinChange.py
class Solution:
    def change(self, amount: int, coins: List[int]) -> int:
        
        ans = [1] + [0]*amount
        
        for coin in coins:
            for i in range(coin, amount+1):
                ans[i] += ans[i-coin]
                
        return ans[-1]
        

File: 2020 Leetcode Challenges/06 June Leetcode Challenge 2020/07 coinChange.cpp
#include<bits/stdc++h>
using namespace std;
  
int change(int amount, vector<int>  coins){
    
    int ways[amount+1];
    ways[0] = 1;
    
    for (int coin : coins){
        for (int i = coin; i <= amount; ++i){
            ways[i] += ways[i-coin];
        }
    }
    
    return ways[amout];  
  }
